06-working-w-copilot/copilot-in-projects.md

```diff
@@ -0,0 +1,980 @@
+
+## Check for Understanding
+
+What are things we need to look out for when using the `/tests` shortcut?
+The tests being placed in the source file
+If Copilot brought in the right imports when referencing other classes or files
+The test file being generated in the source folder instead of test folder
+Whether the test assertions are assuming the correct kind of data
+
+<!-- Question 1 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: checkbox
+* id: 9100b9a2-b806-4b0c-94d2-5f1af5acf54b
+* title: Using `/tests`
+
+##### !question
+Select the options below that we need to look out for when using the `/tests` shortcut.
+##### !end-question
+
+##### !options
+* The tests being placed in the source file when accepted
+* If Copilot brought in the right imports when referencing other classes or files
+* A test file being generated in the source folder instead of test folder
+* Whether the test assertions are assuming the correct kind of data
+##### !end-options
+
+##### !answer
+* If Copilot brought in the right imports when referencing other classes or files
+* A test file being generated in the source folder instead of test folder
+* Whether the test assertions are assuming the correct kind of data
+##### !end-answer
+
+### !end-challenge
+<!-- prettier-ignore-end --> 
+
+<!-- Question 2 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: checkbox
+* id: b53afd64-55e1-4670-8d3b-bce1ca3b8872
+* title: Refactoring with Copilot Concerns
+
+##### !question
+Select the refactoring steps below where Copilot may **not** be useful. 
+##### !end-question
+
+##### !options
+* Gathering requirements
+* Locating Dependencies
+* Checking and expanding our test suite
+* Executing the code change
+##### !end-options
+
+##### !answer
+* Gathering requirements
+* Locating Dependencies
+##### !end-answer
+
+##### !explanation
+* Gathering requirements - Copilot has no way of knowing what it is we want to build and requirements like speed, memory, etc.; those need to be decided by people, ideally before we start building the software. 
+* Locating Dependencies - Copilot doesn't automatically have access to every file in a project, so it can easily miss dependencies. 
+##### !end-explanation
+
+### !end-challenge
+<!-- prettier-ignore-end -->
```

Comment:
First look - Overall, I think the content looks pretty great for this lesson! You cover a lot of bases and go into plenty of detail. Your big ask was to see if this lesson needs to be split up, and I definitely think that's the case. In my view, I could see this lesson definitely being split into two lessons. 

As is, the lesson is divided up as follows:
- Goals
- Writing New code
- Writing the Author, Genre and BookGenre Models
- Refactoring

I think the most obvious break is to make the refactoring portion and everything after it be its own lesson. That's probably the heftiest portion of this overall lesson and gives a nice split between working with new code vs working with existing code. One thing I especially like about this is that the common theme of testing is woven between the two sections so the students get to have that thread connecting the two lessons. 

I will say, I could honestly see an argument for also splitting up Writing New Code and Writing the Author, Genre and BookGenre models as well. This gives us three overarching sections of "Writing New Code", "Writing New Code based off existing code" and "Refactoring existing Code". That might make things a little more manageable for our students. That being said, this is just my first glance. I'm going to revisit this in a bit to see if I notice anything else, but I wanted to get you a first pass of feedback!

Reply:
Thank you! For the moment I've split the Copilot in projects into 2 sections to see how that flows.